src/plugins/apm.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020-present, Elastic NV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { randomBytes } from 'crypto';
+import { Request, Route } from 'playwright-core';
+import { ApmOptions, Driver } from '../common_types';
+
+export class Apm {
+  constructor(private driver: Driver, private options: ApmOptions) {}
+
+  async traceHandler(route: Route, request: Request) {
+    const traceId = randomBytes(16).toString('hex');
+    // We dont want to track Synthetics, so sending a dummy id
+    const parentId = '0'.repeat(16);
+    const traceparent = `00-${traceId}-${parentId}-01`;
+    const headers = {
+      ...(await request.allHeaders()),
+      traceparent,
```

Comment:
Can you send the `monitor.id` and `execution.id` in the `tracestate` header? Then I can talk to the APM agent devs to persist this field on APM docs. 

This will make it possible to to retrieve Synthetic monitors for a particular APM service.

Something like this:
```suggestion
      traceparent,
      tracestate: `synthetics.monitor.id=1234,synthetics.execution.id=5678`
```

See more about trace-state here: https://www.elastic.co/guide/en/apm/agent/rum-js/current/distributed-tracing-guide.html#enable-tracestate

Reply:
Just adding it here as well: there is an ongoing discussion whether to use `tracestate` or the `baggage`  header instead. `baggage` seems to be a better fit. Discussion is linked internally.

Reply:
Thanks for chiming in @gregkalapos. I've updated my suggestion to use `baggage` instead of `tracestate`

Reply:
One more change: the synthetics agent should only send `baggage` header - not the `traceparent` header. The reason is that if the synthetics client start a trace (by generating a trace id) but doesn't send a transaction (like the RUM agent would) the resulting trace will be "root-less because it does not have an entry transaction.

By not sending `traceparent` the receiving service will instead create the `trace.id` and will also send up the entry transaction.